test/relayHub/RelayHub.test.ts

```diff
@@ -1,1177 +1,507 @@
+            await addWorker(relayManager);
 
-      it('should NOT be able to add workers when Manager is unstaked', async function () {
-        const promiseOfWorker = mockRelayHub
-          .connect(relayManager)
-          .addRelayWorkers([relayWorkerAddr]);
+            // Enable more than one worker per manager
+            await mockRelayHub.setVariable('maxWorkerCount', 2);
+          });
 
-        await expect(promiseOfWorker).to.have.revertedWith(
-          ERR_UNSTAKED
-        );
-      });
+          it('shoud map the manager addr onto the worker addr', async function () {
+            const expectedManagerAddr = relayManagerAddr;
+            const actualManager = await mockRelayHub.workerToManager(
+              relayWorkerAddr
+            );
+
+            const actualManagerAddr = ethers.utils.getAddress(
+              `0x${actualManager.slice(25, actualManager.length - 1)}`
+            );
+
+            expect(actualManagerAddr).to.eq(expectedManagerAddr);
+          });
+
+          it('should NOT register the same worker twice', async function () {
+            // Adding same worker for the same manager second time
+            await expect(addWorker(relayManager)).to.be.revertedWith(
+              ERR_WORKER_HAS_MANAGER
+            );
+          });
+
+          it('should NOT register the same worker for different managers', async function () {
+            const [anotherRelayManager] = otherUsers;
 
-      describe('with one worker', function() {
-        const addWorker = (
-          manager: SignerWithAddress,
-          workerAddr = relayWorkerAddr
-        ) => mockRelayHub.connect(manager).addRelayWorkers([workerAddr]);
+            await mockRelayHub.stakeForAddress(
+              anotherRelayManager.address,
+              1000,
+              {
+                value: oneRBTC,
+              }
+            );
 
-        beforeEach(async function() {
-          // Add stake to manager
+            await expect(addWorker(anotherRelayManager)).to.be.revertedWith(
+              ERR_WORKER_HAS_MANAGER
+            );
+          });
+
+          it('should fail when adding more workers than maxWorkerCount', async function () {
+            const expectedWorkerCountLimit = 13;
+            await mockRelayHub.setVariable('maxWorkerCount', expectedWorkerCountLimit);
+
+            await mockRelayHub.stakeForAddress(relayManagerAddr, 1000, {
+              value: oneRBTC.mul(expectedWorkerCountLimit),
+            });
+
+            const workers = Array.from(Array(expectedWorkerCountLimit)).map(
+              () => ethers.Wallet.createRandom().address
+            );
+            const promiseOfWorkers = mockRelayHub
+              .connect(relayManager)
+              .addRelayWorkers(workers);
+
+            await expect(promiseOfWorkers).to.revertedWith(ERR_TOO_MANY_WORKERS);
+          });
+        });
+      });
+
+      describe('disableRelayWorkers', function () {
```

Comment:
There are some cases missing:
-"invalid quantity of workers"
-Check if Relay Manager is staked
-Disable more than one worker
-"Incorrect Manager"
-Disable a subset of the workers.

Reply:
Added as to do. Good catch.